src/main/java/org/mitre/synthea/engine/Transition.java

```diff
@@ -283,13 +283,13 @@ public String follow(Person person, long time) {
         } else if (currentAttribute.equalsIgnoreCase("time")) {
           // do nothing, we already have it
         } else {
-          String personsAttribute
-              = (String) person.attributes.get(currentAttribute.toLowerCase());
-          if (personsAttribute == null) {
+          if (!person.attributes.containsKey(currentAttribute)) {
             throw new RuntimeException("LOOKUP TABLE ERROR: Attribute '"
                 + currentAttribute + "' in CSV table '" + this.lookupTableName
                 + "' does not exist as one of this person's attributes.");
           }
+          String personsAttribute
+              = person.attributes.get(currentAttribute).toString();
```

Comment:
Question on using `toString()` here... Is this the behavior we want?

It could be "dangerous" as an attribute that is not a `String` in the `attributes` map, but some other object, could get "flattened" into a `String`. Would it be better to cast to `String` and potentially crash if the object is not a `String`?

Reply:
This is what I get for pulling in features that I didn't write myself. 😁  As I look closer at this now, I think I misread where things would be different before vs after.  (This was the original commit on `cabg_cardio_gmf` btw: https://github.com/synthetichealth/synthea/commit/8f2a471347982462d0d0659ce3abdfc335df64f3)

The real difference is the removal of the `toLowerCase()` on the attribute name. There's not much changed on nulls, because previously the explicit null check would throw an exception, now a NullPointerException will be thrown on the `toString`. I don't mind the "flattening to a string" aspect too much, because I would expect it to be useful in cases like booleans and integers. Seems like Jay was using booleans in lookup tables in that original commit. So at this point I think the options are a) leave the toString (or a_ii is to replace it with `String.valueOf()` to intentionally convert `null` into `"null"`), or b) consider a list of types that are allowed to be here, and crash if it's not one of those types. Then that immediately raises the question of should floating point numbers be allowed? Comparing floats as strings sounds like a bad idea, but I think in many cases we use `double`s to be more generic, but they actually contain a whole number.

I feel like Jay's preference is to generally not crash. I would lean toward leaving the toString more because it might allow future things to work with fewer code changes.

Reply:
Yes, Booleans in the table-based transition table. Casting the attribute as a (String) was causing exceptions.

And, in the end, the attribute needs to be a String, because of how the LookupTableKey class works. 

This change intended to make the table-based transitions function for boolean data, without refactoring the entire implementation (which I didn't have time for).

Reply:
Makes sense.